optimum/exporters/onnx/model_configs.py

```diff
@@ -2621,3 +2621,23 @@ class EncoderDecoderOnnxConfig(EncoderDecoderBaseOnnxConfig):
     NORMALIZED_CONFIG_CLASS = NormalizedEncoderDecoderConfig
 
     DEFAULT_ONNX_OPSET = 14  # uses SDPA in Transformers, hence opset>=14.
+
+
+class GITOnnxConfig(VisionOnnxConfig):
+    NORMALIZED_CONFIG_CLASS = NormalizedVisionConfig
```

Comment:
The issue you're reporting `ValueError: Input image size (64*64) doesn't match model (32*32).` should be fixed if you replace the config with :

```suggestion
    NormalizedTextAndVisionConfig.with_args(vision_config="vision_config")
```

as it seems that for  GitConfig the `image_size` attribute needs to be taken from the `vision_config` directly https://github.com/huggingface/transformers/blob/504c4d36929b6bb8a8c2ecfad0f2625f4075f22a/src/transformers/models/git/configuration_git.py#L98.

What is currently happening is that before export this value is not found in the config and is default to 64 https://github.com/huggingface/optimum/blob/d21256c2964945fc3fe4623f7befb21082b69a25/optimum/utils/input_generators.py#L805 when it should be set to 32 in your case  https://huggingface.co/hf-internal-testing/tiny-random-GitModel/blob/main/config.json#L52